components/calendar/src/hebrew.rs

```diff
@@ -182,17 +220,42 @@ impl Calendar for Hebrew {
             }
         };
 
-        ArithmeticDate::new_from_ordinals(year, month_ordinal, day).map(HebrewDateInner)
+        ArithmeticDate::new_from_ordinals_with_info(year, month_ordinal, day, year_info)
+            .map(HebrewDateInner)
     }
 
     fn date_from_iso(&self, iso: Date<Iso>) -> Self::DateInner {
         let fixed_iso = Iso::fixed_from_iso(*iso.inner());
-        Self::civil_hebrew_from_fixed(fixed_iso).inner
+        let (year_info, h_year) = YearInfo::year_containing_rd(fixed_iso);
+        // Obtaining a 1-indexed day-in-year value
+        let mut day = fixed_iso - year_info.new_year() + 1;
+
+        let year_info = HebrewYearInfo::compute_with_keviyah(year_info.keviyah, h_year);
+        for month in 1..14 {
```

Comment:
Thought: Seems like this loop should be on the Keviyah type since it is essentially the inverse of `days_preceding`

Reply:
I added this, though I'm not sure how to test this with the existing test. I may instead write a second roundtrip test that picks a handful of years but tests every day in that year.